lib/gc/Analysis/MatmulConfigAnalysis.cpp

```diff
@@ -352,102 +384,110 @@ bool readConfigFromAttrs(MatmulConfig &config, ArrayRef<NamedAttribute> attrs) {
+        givenInnermostBlock.push_back(config.innerMostMBlock);
+      } else {
+        givenInnermostBlock.push_back(0);
+      }
+      if (NDimTypeIdx.size() > 1) {
+        config.innerMostNBlock = 1;
+        for (auto &&[i, d] : llvm::enumerate(NDimTypeIdx))
+          if (i != 0)
+            config.innerMostNBlock *=
+                linalgOp.getShape(linalgOp.getDpsInputOperand(1))[d];
+        givenInnermostBlock.push_back(config.innerMostNBlock);
+      } else {
+        givenInnermostBlock.push_back(0);
+      }
+      if (KDimTypeIdx.size() > 1) {
+        config.innerMostKBlock = 1;
+        for (auto &&[i, d] : llvm::enumerate(KDimTypeIdx))
+          if (i != 0)
+            config.innerMostKBlock *=
+                linalgOp.getShape(linalgOp.getDpsInputOperand(1))[d];
+        givenInnermostBlock.push_back(config.innerMostKBlock);
+      } else {
+        givenInnermostBlock.push_back(0);
+      }
 
-    LLVM_DEBUG(llvm::dbgs()
-               << "M: " << M << ", N: " << N << ", K: " << K << "\n");
+      LLVM_DEBUG(llvm::dbgs()
+                 << "M: " << M << ", N: " << N << ", K: " << K << "\n");
 
-    // try to read the config from the attributes
-    SmallVector<NamedAttribute> attrs(linalgOp->getAttrs());
-    bool hasPredefinedConfig = readConfigFromAttrs(config, attrs);
+      // try to read the config from the attributes
+      SmallVector<NamedAttribute> attrs(linalgOp->getAttrs());
+      bool hasPredefinedConfig = readConfigFromAttrs(config, attrs);
 
-    // if there is a given config, skip the cost model
-    if (!hasPredefinedConfig) {
-      LLVM_DEBUG(llvm::dbgs() << "No predefined config\n");
-      // TODO: Could add a weight or priority for cost model
-      SmallVector<std::tuple<CostModelFn, std::string, double>> costModelList =
-          {{workloadBalancedCost, "workloadBalancedCost", 1},
-           {vectorRegEfficiencyCost, "vectorRegEfficiencyCost ", -1},
-           {computationIntensityOnL2Cache, "computationIntensityOnL2Cache", -1},
-           {memoryConsumptionOnThreadCost, "memoryConsumptionOnThreadCost",
-            -1}};
-      SmallVector<uint32_t> shape = {M, N, K};
-      std::vector<MatmulConfig> configCandidates =
-          prepareConfigCandidates(root, sysDesc, shape, givenInnermostBlock);
-      for (auto &&[fn, name, threshold] : costModelList)
-        configCandidates = filterConfigByCostModel(
-            configCandidates, linalgOp, shape, sysDesc, fn, 0.5, threshold);
-      if (!configCandidates.empty())
-        config = configCandidates[0];
-    }
+      // if there is a given config, skip the cost model
+      if (!hasPredefinedConfig) {
+        LLVM_DEBUG(llvm::dbgs() << "No predefined config\n");
+        // TODO: Could add a weight or priority for cost model
+        SmallVector<std::tuple<CostModelFn, std::string, double>>
+            costModelList = {
+                {dynamicBufferizationCost, "dynamicBufferizationCost", 0.9},
```

Comment:
Can you add a simple notes regarding the `weight/priority` used here?

Reply:
A note is added